app/views/play/level/PlayWebDevLevelView.coffee

```diff
@@ -12,10 +13,18 @@ module.exports = class PlayWebDevLevelView extends RootView
   id: 'play-web-dev-level-view'
   template: require 'templates/play/level/play-web-dev-level-view'
 
-  initialize: (@options, @levelID, @sessionID) ->
+  initialize: (@options, @sessionID) ->
     @courseID = utils.getQueryVariable 'course'
-    @level = @supermodel.loadModel(new Level _id: @levelID).model
     @session = @supermodel.loadModel(new LevelSession _id: @sessionID).model
+    @level = new Level()
+    @session.once 'sync', =>
+      levelResource = @supermodel.addSomethingResource('level')
+      api.levels.getByOriginal(@session.get('level').original).then (levelData) =>
+        @levelID = levelData.slug
+        @level.set({ _id: @levelID })
+        @level.fetch()
+        @level.once 'sync', =>
+          levelResource.markLoaded()
```

Comment:
This is a bit over-complicated by SomethingResource because we can't just chain promises; the supermodel marks itself loaded after just the first model if we do.